application/service/consumer_service.py

```diff
@@ -298,7 +300,18 @@ def check_and_update_block_confirmation(self, tx_id, blockchain_name, required_b
         if blockchain_name == BlockchainName.CARDANO.value.lower():
             if required_block_confirmation > current_block_confirmation:
                 logger.info("Block confirmation is not enough to consider, so checking the block confirmation again")
-                current_block_confirmation = get_block_confirmation(tx_hash=tx_hash, blockchain_network_id=network_id)
+                i = 1
+                while not current_block_confirmation:
+                    current_block_confirmation = get_block_confirmation(tx_hash=tx_hash,
+                                                                        blockchain_network_id=network_id)
+                    if BLOCK_CONFIRMATION_SLEEP_TIME:
```

Comment:
Shouldnt we break when we get a confirmation? The if condition should be 
`if not current_block_confirmation`

Reply:
thanks sir